domain/application/state/application.go

```diff
@@ -1240,6 +1268,147 @@ ON CONFLICT(application_uuid) DO UPDATE
 	return errors.Annotatef(err, "updating cloud service for application %q", name)
 }
 
+type statusKeys []string
+
+// saveStatusData saves the status key value data for the specified unit in the specified table.
+// It's called from each different SaveStatus method which previously has confirmed the unit UUID exists.
+func (st *ApplicationState) saveStatusData(ctx context.Context, tx *sqlair.TX, table, unitUUID string, data map[string]string) error {
+	unit := coreUnit{ID: unitUUID}
```

Comment:
I think the name for this type is poor.

Reply:
Naming is hard. The struct holds just "core" or minimal attributes of a unit - uuid, name, life, net node uuid. Maybe "minimalUnit". I'll have a think.

Reply:
coreUnit is terrible because it clashes with similarly named coreunit (which you have as an aliased package in this very file) and the concept of "core" when it comes to juju.

Something like:
- `basicUnit`
- `minimalUnit`
- `unitIntrinsics`

IDK. I wouldn't think about it to hard, it is an unexported concept.

Reply:
I already changed to `minimalUnit`